### PR TITLE
Non breaking `CaseIterable` conformance

### PR DESCRIPTION
**Pull request by gjcairo**
This is part of a wider effort to make breaking changes introduced in main non-breaking.

The `SwiftProtobuf.Enum` protocol started inheriting from `CaseIterable`. This is a breaking change, since it's a new requirement of the protocol to make sure all conforming concrete types also conform to `CaseIterable`. 

To allow conformance of generated enums to `CaseIterable` without being source-breaking, this PR modified the code generator to conform all types conforming to `Enum` to `CaseIterable`, but removes the inheritance of `CaseIterable` from the `Enum` `protocol`.

A separate commit regenerates the protos, and a third one regenerates the references for tests.

### CODE REVIEW

File: Sources/protoc-gen-swift/EnumGenerator.swift [hunk 1]
```diff
@@ -22,6 +22,8 @@ private let unrecognizedCaseName = "UNRECOGNIZED"
 
 /// Generates a Swift enum from a protobuf enum descriptor.
 class EnumGenerator {
+  private static let requiredProtocolConformancesForEnums = ["Swift.CaseIterable"].joined(separator: ", ")
```
Comment (thomasvl) [inline Sources/protoc-gen-swift/EnumGenerator.swift:25]: Should we put a todo here to move this onto enum when we can do breaking changes?

**Review by thomasvl — APPROVED** [review]

**Review by thomasvl — APPROVED** [review]

**Review by FranzBusch — APPROVED** [review]

**Comment by thomasvl** [comment]
Looks like you might need to sync/rebase and rerun the regeneration & reference steps to get a few more things updated.

**Comment by gjcairo** [comment]
@thomasvl I'm getting the following error when running `make regenerate`: 
```
conformance/test_protos/test_messages_edition2023.proto: This file uses editions, but --experimental_editions has not been enabled. This syntax is experimental and should be avoided.
```
File was added [here](https://github.com/apple/swift-protobuf/commit/0c66ecd4b050c298d2916de9746856d8cee25326#diff-795344aad14aa33c371e81962745206038bb30b5844e0ae47a6888f199cb0fdf). My `protoc` version is 26.1 (downloaded from https://github.com/protocolbuffers/protobuf/releases/tag/v26.1

Am I missing something, or should we change the command we run from the Makefile?

**Comment by thomasvl** [comment]
Use a `main` of protobuf to get the updated editions support.

**Comment by gjcairo** [comment]
@thomasvl thanks - tests are passing now

**Comment by thomasvl** [comment]
> @thomasvl thanks - tests are passing now

That's was check about to check.  Merging.

**Pull request merged**